lib/list_versions.go

```diff
@@ -126,17 +109,71 @@ func GetTFURLBody(mirrorURL string) ([]string, error) {
 		os.Exit(1)
 	}
 
-	body, errBody := ioutil.ReadAll(resp.Body)
-	if errBody != nil {
-		log.Printf("[Error] : reading body: %v", errBody)
+	body := new(bytes.Buffer)
+	if _, err := io.Copy(body, resp.Body); err != nil {
+		log.Fatal(err)
+	}
+	if err := json.Unmarshal(body.Bytes(), &releases); err != nil {
+		log.Fatalf("%s: %s", err, body.String())
+	}
+	return releases, nil
+}
+
+func GetTFReleases(mirrorURL string, preRelease bool) ([]Release, error) {
+	// temp testing
+	queryParams := map[string]string{"limit": "20"}
+	rel, _ := getReleases(mirrorURL, queryParams)
+
+	var releases []Release
+	rel, _ = getReleases(mirrorURL, queryParams)
+	releases = append(releases, rel...)
+	for len(rel) == 20 {
+		queryParams["after"] = rel[len(rel)-1].TimestampCreated.String()
+		rel, _ = getReleases(mirrorURL, queryParams)
+		releases = append(releases, rel...)
+	}
+
+	if !preRelease {
+		for i, r := range releases {
+			if r.IsPrerelease {
+				releases = removePreReleases(releases, i)
+			}
+		}
+	}
+	sort.Slice(releases, func(i, j int) bool {
+		return releases[i].Version > releases[j].Version
+	})
+	return releases, nil
+}
+
+func GetTFRelease(mirrorURL, requestedVersion string) (Release, error) {
+	resp, errURL := httpGet(mirrorURL+"/"+requestedVersion, nil)
+	if errURL != nil {
+		log.Printf("[Error] : Getting url: %v", errURL)
+		os.Exit(1)
+		return Release{}, errURL
```

Comment:
As I wrote above I'm not quite familiar with Golang hence the question: how does this work — shouldn't `os.Exit()` abort execution right away?
What I mean is that will it indeed get to the `return` statement at all?

Reply:
Yeah it does exactly that. I didn't add that code but will clean this up too.

Reply:
Thanks @Jukie 🙇🏻 Sorry I didn't get to figure out which code was added by you and which you didn't add 🤷🏻